index.js

```diff
@@ -0,0 +1,105 @@
+const { PrismaClient } = require('@prisma/client')
+const { v4: uuidv4 } = require("uuid")
+
+const prisma = new PrismaClient()
+
+// retryTxn wraps SQL operations in an explicit transaction. 
+// If the transaction fails, the function retries the operation, with exponential backoff.
+async function retryTxn(n, max, operation) {
+    while (true) {
+        n++
+        if (n === max) {
+            throw new Error("Max retry count reached.")
+        }
+        try {
+            await operation()
+            await prisma.$queryRaw`COMMIT;`
+            return
+        } catch (err) {
+            if (err.code !== "40001") {
```

Comment:
can you add a comment about what this error code is?

Reply:
Done.